src/wayland/content_type/mod.rs

```diff
@@ -132,14 +132,14 @@ impl ContentTypeSurfaceData {
 
 /// User data of `WpContentTypeV1` object
 #[derive(Debug)]
-pub struct ContentTypeUserData(Mutex<Option<WlSurface>>);
+pub struct ContentTypeUserData(Mutex<WlSurface>);
```

Comment:
From the protocol:
>  When the associated surface gets destroyed, this object becomes inert and the client should destroy it.

Given the *should* in there, I believe this should be a `Weak<WlSurface>` to not prevent the surface user data from being cleaned up.

Reply:
Yep, sounds right, done